### ISSUE TITLE
Warnings generated when run against the current SPDX 3 Model

### ISSUE THREAD

**Issue by goneall** [spdx/spec-parser#142]
The following warnings are generated when running against the SPDX 3 model:

```
C:\Users\gary\git\spec-parser\venv\Lib\site-packages\rdflib\plugins\serializers\rdfxml.py:280: UserWarning: Assertions on rdflib.term.BNode('Nff3935e9f1bc44e4b727a01139b2bb3f') other than RDF.first and RDF.rest are ignored ... including RDF.List
  self.predicate(predicate, object, depth + 1)
```

Attached is a zip of the model directory: 
[model.zip](https://github.com/user-attachments/files/16802832/model.zip)

**Comment by zvr** [comment]
Yes, it's a RDFlib limitation (RDFlib is the Python library used for handling RDF).
One of the serializers (the "prettyXML" one) does not handle assertions other than `first` and `rest` (as it says).

You can check it on the source code upstream at https://rdflib.readthedocs.io/en/stable/_modules/rdflib/plugins/serializers/rdfxml.html
(search for `# Warn that any assertions on object` to get the exact line)

This started appearing when we introduced blank nodes (I _think_ all warnings are about blank nodes).

I have not had the time to compare all different serialization to establish what is actually missing and how much of a problem it is in practice.

**Comment by goneall** [comment]
> I have not had the time to compare all different serialization to establish what is actually missing and how much of a problem it is in practice.

I have not seen any issue myself in the produced model artifacts.

**Comment by zvr** [comment]
OK, after further research, my [comment above](https://github.com/spdx/spec-parser/issues/142#issuecomment-2322830736) still holds.

The outcome is that the ontology generated by that specific serializer (the "prettyXML" one) does not contain the RDF that specifies that a property of a Vocabulary type has to take only one the values of the Vocabulary.

Presumably this will be fixed in RDFlib (upstream) at some point.

**Comment by zvr** [comment]
Closing this as nothing to do here; waiting for RDFlib fix.
